setup.py

```diff
@@ -229,37 +380,32 @@ def setup_package():
         install_requires=build_requires,
     )
 
-    if len(sys.argv) >= 2 and ('--help' in sys.argv[1:] or
-            sys.argv[1] in ('--help-commands', 'egg_info', '--version',
-                            'clean')):
-        # For these actions, NumPy is not required.
-        #
-        # They are required to succeed without Numpy for example when
-        # pip is used to install Scipy when Numpy is not yet present in
-        # the system.
-        try:
-            from setuptools import setup
-        except ImportError:
-            from distutils.core import setup
-
-        FULLVERSION, GIT_REVISION = get_version_info()
-        metadata['version'] = FULLVERSION
+    if "--force" in sys.argv:
+        run_build = True
     else:
-        if (len(sys.argv) >= 2 and sys.argv[1] in ('bdist_wheel', 'bdist_egg')) or (
-                    'develop' in sys.argv):
-            # bdist_wheel/bdist_egg needs setuptools
-            import setuptools
+        # Raise errors for unsupported commands, improve help output, etc.
+        run_build = parse_setuppy_commands()
 
+    from setuptools import setup
```

Comment:
It's not about commands, because this import will always be executed. But the removing of ``MANIFEST`` needs to be done first, and I think also the ``sdist`` import from ``distutils`` needs to be done before the monkeypatching madness starts.